AppKit/CPSearchField.j

```diff
@@ -773,6 +769,63 @@ var CPAutosavedRecentsChangedNotification = @"CPAutosavedRecentsChangedNotificat
 
 @end
 
+#pragma mark -
+
+@implementation CPSearchField (CPTrackingArea)
+{
+    CPTrackingArea      _searchButtonTrackingArea;
```

Comment:
We do not declare instance variable in categories, neither does objective-c. I heard from @mrcarlberg that the compiler tolerates it but for consistency these should go in the  main class implementation.

Reply:
Well, in fact, I was considering this more like a class extension... (thus permitting declaration of properties and instance variables). This way, the code is really isolated and, but that's my own feeling, is easier to read / understand. Also, when adding new things (like use of tracking areas or CSS theming), the original code remains untouched.
But, OK, if it was a true class extension, I wouldn't have put "(CPTrackingArea)" (I've tried, of course found that we do not support class extension, and also considered that it was more clear to have a kind of "subject", like for a category).

Are you sure you prefer not to use this ?

Reply:
(I love to isolate / structure / ... the code as it is easier to understand / maintain ;-) )

Reply:
I kind of agree with you, it would be a cool feature to have on objective-j. But we can't experiment it just in one place. It has to be tested, validated and documented. If you want, you could open a new issue to discuss that.

Reply:
I think it is ok to use this feature. In Objective-C you can declare a property in categories. Objective-J does not support property declarations anywhere. We do support instance variable in a category and it is not supported in Objective-C as @cacaodev said above.

You could say that if you want a property in a category in Objective-J you have to declare it as an instance variable.